src/Common/unix/FileStream_unix.cpp

```diff
@@ -1,5 +1,31 @@
 #include "Common/unix/FileStream_unix.h"
 
+std::optional<fs::path> findPathCI(const fs::path& path)
+{
+	if (fs::exists(path)) return path;
+	if (!path.has_parent_path()) return {};
+	auto parrentPath = path.parent_path();
+	if (!fs::exists(parrentPath))
+	{
+		auto tempParrentPath = findPathCI(parrentPath);
+		if (tempParrentPath.has_value())
+			parrentPath = std::move(tempParrentPath.value());
+		else
+			return {};
+	}
+	std::string fName = path.filename().string();
+	if (fs::exists(parrentPath))
+		for (auto&& dirEntry : fs::directory_iterator(parrentPath))
+		{
+			std::string dirFName = dirEntry.path().filename().string();
+			if (boost::iequals(dirFName, fName))
```

Comment:
Does this work with non-ascii strings?

Reply:
If the filename contains non-ASCII characters, e.g. "fileỰ.txt", and the case matches for those characters, e.g. "FiLeỰ.TxT", then it works. But it will not work if it tries to look for "fileự.txt".
I could make it work for non-ASCII file using Boost.Locale if it's necessary.